pyquil/quiltwaveforms.py

```diff
@@ -1,141 +1,65 @@
-from copy import copy
-from dataclasses import dataclass
-from numbers import Complex, Real
-from typing import Callable, Dict, Union, List, Optional, no_type_check
+from typing import Optional
```

Comment:
So this reaches way back, but I think there's a chance this file / these waveform classes are not used anywhere. Matt pointed out a critical bug here maybe a year after they were released and it hadn't caused any trouble that we could discover or had heard about.


It's great that you did everything necessary to preserve them and make them work, so we don't have to break anything if I have that wrong, or if people have started using them since then. Let's also mark everything here as deprecated as a sort of survey for users. If people speak up about this, we can un-deprecate; if not, we know it's good to clear out in v5.

Reply:
Sounds good, I'll deprecate the module + the `TemplateWaveform` base class.